src/ring_theory/polynomial/cyclotomic/eval.lean

```diff
@@ -160,4 +184,311 @@ begin
     exact nat.pow_right_injective hp.two_le hxy }
 end
 
+open_locale complex_conjugate nnreal
+
+section is_R_or_C
+open is_R_or_C
+
+lemma eq_neg_iff {R : Type*} [ring R] [invertible (2 : R)] (x : R) :
+  x = -x ↔ x = 0 :=
+begin
+  have : is_unit (⅟ (2:R)) := is_unit_of_invertible _,
+  rw [eq_neg_iff_add_eq_zero, ← two_mul, ← this.mul_right_inj, inv_of_mul_self_assoc, mul_zero],
+end
+
+lemma conj_eq_neg_iff {K : Type*} [is_R_or_C K] (z : K) :
+  conj z = -z ↔ re z = 0 :=
+begin
+  rw [← re_add_im z, ring_hom.map_add, ring_hom.map_mul, conj_of_real, conj_of_real, conj_I,
+      neg_add, re_add_im, mul_neg, add_left_inj, eq_neg_iff, of_real_eq_zero],
+end
+end is_R_or_C
+
+@[simp]
+lemma conj_exp_primitive_root (n : ℕ) : conj (complex.exp (2 * ↑real.pi * complex.I / ↑n)) =
+                                        complex.exp (- (2 * ↑real.pi * complex.I / ↑n)) :=
+begin
+  rw ← complex.exp_conj,
+  congr,
+  simp only [div_eq_mul_inv, is_R_or_C.re_to_complex, complex.mul_re, complex.bit0_im, complex.I_re,
+             complex.one_im, bit0_zero, complex.of_real_im, mul_zero, sub_zero, conj_eq_neg_iff,
+             complex.mul_im, zero_mul, add_zero, complex.inv_im, complex.nat_cast_im, neg_zero],
+end
+
+lemma units.mk0_prod {β α : Type} [_inst_1 : comm_group_with_zero β] (s : finset α)
```

Comment:
Idem

Reply:
@alexjbest, everything except this lemma I deleted in a commit I'll push soon. If you want those lemmas, feel free to PR them, but I don't think they're needed for the proof anymore and I'm getting serious PR fatigue

Reply:
No problem, thanks for letting me know and i understand completely!
I might take a look and PR them but if they aren't needed they certainly aren't your responsibility, I've added a lot of work with my requests to your pr already!